examples/inverse/plot_morph_stc_to_fsaverage_dipy.py

```diff
@@ -0,0 +1,231 @@
+t1_m = nib.load(t1_fname)
+
+# get Moving to world transform
+t1_m_grid2world = t1_m.affine
+
+# output Moving as ndarray
+t1_m = t1_m.dataobj[:, :, :]
+
+# normalize values
+t1_m = t1_m.astype('float') / t1_m.max()
+
+# load fsaverage brain (Static)
+t1_fname = data_path + '/subjects/fsaverage/mri/brain.mgz'
+t1_s = nib.load(t1_fname)
+
+# get Static to world transform
+t1_s_grid2world = t1_s.affine
+
+# output Static as ndarray
+t1_s = t1_s.dataobj[:, :, :]
+
+# normalize values
+t1_s = t1_s.astype('float') / t1_s.max()
+
+##############################################################################
+# Compute affine transformation using mutual information metric
+
+# compute center of mass
+c_of_mass = transform_centers_of_mass(t1_s, t1_s_grid2world,
+                                      t1_m, t1_m_grid2world)
+
+nbins = 32
+
+# prepare affine registration
+affreg = AffineRegistration(metric=MutualInformationMetric(nbins, None),
+                            level_iters=[10000, 1000, 100],
+                            sigmas=[3.0, 1.0, 0.0],
+                            factors=[4, 2, 1])
+
+# translation
+translation = affreg.optimize(t1_s, t1_m, TranslationTransform3D(), None,
+                              t1_s_grid2world, t1_m_grid2world,
+                              starting_affine=c_of_mass.affine)
+
+# rigid body transform (translation + rotation)
+rigid = affreg.optimize(t1_s, t1_m, RigidTransform3D(), None,
+                        t1_s_grid2world, t1_m_grid2world,
+                        starting_affine=translation.affine)
+
+# affine transform (translation + rotation + scaling)
+affine = affreg.optimize(t1_s, t1_m, AffineTransform3D(), None,
+                         t1_s_grid2world, t1_m_grid2world,
+                         starting_affine=rigid.affine)
+
+# apply affine transformation
+t1_m_affine = affine.transform(t1_m)
+
+##############################################################################
+# Compute Symmetric Diffeomorphic Registration
+
+# set up Symmetric Diffeomorphic Registration (metric, iterations per level)
+sdr = SymmetricDiffeomorphicRegistration(CCMetric(3), [10, 10, 5])
+
+# compute mapping
+mapping = sdr.optimize(t1_s, t1_m_affine)
+
+##############################################################################
+# Apply transformations and plot
+
+# morph stc data
+stc_sdr_affine = mapping.transform(affine.transform(stc))
+
+# plot result
+slice_sel = 100
+
+fig = plt.figure()
+plt.subplot(131)
+plt.imshow(t1_m[slice_sel, :, :] + stc[slice_sel, :, :])
+plt.title('subject brain')
+plt.set_cmap('jet')
+plt.axis('off')
+plt.subplot(132)
+plt.imshow(t1_s[slice_sel, :, :] + stc[slice_sel, :, :])
+plt.title('fsaverage brain\nprior transformation')
+plt.axis('off')
+plt.subplot(133)
+plt.title('fsaverage brain\nafter transformation')
+plt.imshow(t1_s[slice_sel, :, :] + stc_sdr_affine[slice_sel, :, :])
+plt.axis('off')
+plt.suptitle('moving individual source data to fsaverage')
+plt.show()
```

Comment:
Couldn't this be made more pythonic? Looking and object-oriented MPL should save at least half of the lines for a gain in readability.

Reply:
hey @jona-sassenhagen thanks for the feedback! You're totally right ;) - would you like it more like this?

```imgs = [t1_m, t1_s, t1_s]
stcs = [stc, stc, stc_sdr_affine]
titles = ['subject brain', 'fsaverage brain\nprior transformation', 'fsaverage brain\nafter transformation']

fig, ax = plt.subplots(1, 3)
plt.set_cmap('jet')
j = 0
for img, stc, title in zip(imgs, stcs, titles):
    ax[j].imshow(img[slice_sel, :, :] + stc[slice_sel, :, :])
    ax[j].set_title(title)
    ax[j].axis('off')
    j += 1
plt.show()´´´

Reply:
Probably best to loop over axes too, while we're at it.